src/analyses/baseInvariant.ml

```diff
@@ -701,7 +730,28 @@ struct
               | TFloat (fk, _) -> `Float (FD.cast_to fk c)
               | _ -> `Float c
             in
-            update_lval c x c' FD.pretty
+            let st = update_lval c x c' FD.pretty in
+            (* handle special calls *)
+            begin match t with
+            | TFloat (fk, _) ->
+              begin match x with
+                | ((Var v), offs) -> 
+                  if M.tracing then M.trace "invSpecial" "qry Result: %a\n" Queries.ML.pretty (ctx.ask (Queries.TmpSpecial (v, TmpSpecial.Spec.resolve offs))); 
+                  begin match ctx.ask (Queries.TmpSpecial (v, TmpSpecial.Spec.resolve offs)) with
+                  | `Lifted (Ceil (ret_fk, xFloat)) when FD.is_interval c -> inv_exp (`Float (FD.inv_ceil (FD.cast_to ret_fk c))) xFloat st
```

Comment:
Why is the check if it is an Interval needed here?

Reply:
It was needed in a previous implementation. However, now this is handled correctly in the float domain, so the whole `is_interval` is no longer needed and I removed it